chain/network/src/peer_store.rs

```diff
@@ -187,21 +239,149 @@ impl PeerStore {
         store_update.commit().map_err(|err| err.into())
     }
 
-    pub fn add_peers(&mut self, peers: Vec<PeerInfo>) {
+    fn touch(&mut self, peer_id: &PeerId) -> Result<(), Box<dyn std::error::Error>> {
+        if let Some(peer_state) = self.peer_states.get(&peer_id) {
+            let mut store_update = self.store.store_update();
+            store_update.set_ser(ColPeers, &peer_id.try_to_vec()?, peer_state)?;
+            store_update.commit().map_err(|err| err.into())
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Add list of peers into store.
+    /// When verified is true is because we establish direct connection with such peer and know
+    /// for sure its identity. If we receive a list of peers from another node in the network
+    /// by default all of them are unverified.
+    pub fn add_peers(
+        &mut self,
+        peers: Vec<PeerInfo>,
+        verified: TrustLevel,
+    ) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
One way to make this function more sane without much additional work is to make it private and then add two public functions:
```rust
pub fn add_indirect_peers(&mut self, peers: Vec<PeerInfo>) -> Result<(), Box<dyn std::error::Error>>;

pub fn add_trusted_peer(&mut self, peer: PeerInfo, trust_level: TrustLevel) -> Result<(), Box<dyn std::error::Error>>;
```